### PR TITLE
Active power control

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
ActivePowerControl IIDM extension is provided in a freeware jar on Maven central. This  extension is responsible for managing distributed slack parameters. 

**What is the new behavior (if this is a feature change)?**
Code is now open source.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*
Code using ActivePowerControl extension have to migrate to new public extension.

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-extensions/pom.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    <dependencies>
+        <!-- compile dependencies  -->
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-iidm-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-iidm-test</artifactId>
```
Comment (mathbagu) [inline iidm/iidm-extensions/pom.xml]: Do you really need this dependency with compile scope? probably test is OK?

File: iidm/iidm-extensions/pom.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.xmlunit</groupId>
+            <artifactId>xmlunit-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.powsybl</groupId>
```
Comment (mathbagu) [inline iidm/iidm-extensions/pom.xml]: replace com.powsybl by ${project.groupId} for this dependency and the two next dependencies

File: iidm/iidm-extensions/pom.xml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.powsybl</groupId>
+            <artifactId>powsybl-iidm-xml-converter</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (mathbagu) [inline iidm/iidm-extensions/pom.xml]: Add an empty line at the end of this file

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ActivePowerControl.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.Battery;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Injection;
+
+/**
+ * @author Ghiles Abdellah <ghiles.abdellah at rte-france.com>
+ */
+public class ActivePowerControl<T extends Injection> extends AbstractExtension<T> {
```
Comment (mathbagu) [inline iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ActivePowerControl.java]: Is there a way to limit the allowed Injection subclasses? Maybe one day we'll add VSC converter stations?

File: iidm/iidm-extensions/pom.xml (hunk 1 above)
Comment (geofjamg) [reply iidm/iidm-extensions/pom.xml]: Fixed

File: iidm/iidm-extensions/pom.xml (hunk 2 above)
Comment (geofjamg) [reply iidm/iidm-extensions/pom.xml]: Fixed

File: iidm/iidm-extensions/pom.xml (hunk 3 above)
Comment (geofjamg) [reply iidm/iidm-extensions/pom.xml]: Fixed

File: iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ActivePowerControl.java (hunk 4 above)
Comment (geofjamg) [reply iidm/iidm-extensions/src/main/java/com/powsybl/iidm/network/extensions/ActivePowerControl.java]: Without adding a new interface (for me that won't be a good idea), the only way I found is to put generic constructor (based on T) in protected and add 2 public constructors for generator and battery only. See code below.

**Comment by geofjamg** [comment]
What is the best place for extension documentation? 
Is is closely links to computation models (mainly loadflow), so in loadflow section?

**Comment by geofjamg** [comment]
Doc added

**Pull request merged**